File: app/src/main/java/com/tkdev/ironmanstash/infinity_stones/fragments/Operations.java
package com.tkdev.ironmanstash.infinity_stones.fragments;

import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;

import com.tkdev.ironmanstash.infinity_stones.database.StonesDbHelper;
import com.tkdev.ironmanstash.infinity_stones.fragments.details.SingleStone;
import com.tkdev.ironmanstash.infinity_stones.fragments.allstones.InfinityStone;

import java.util.ArrayList;
import java.util.List;

import static com.tkdev.ironmanstash.infinity_stones.database.StonesContract.DETAIL_TABLE;
import static com.tkdev.ironmanstash.infinity_stones.database.StonesContract.DetailStoneEntry.COLUMN_DETAIL_NAME;
import static com.tkdev.ironmanstash.infinity_stones.database.StonesContract.IF_TABLE;
import static com.tkdev.ironmanstash.infinity_stones.database.StonesContract.InfinityStonesEntry.COLUMN_STONE_NAME;
import static com.tkdev.ironmanstash.infinity_stones.database.StonesContract.InfinityStonesEntry.COLUMN_STONE_VISIBILITY;

public class Operations {

    private static Operations operations;
    private SQLiteDatabase database;
    private Context context;

    public static Operations get(Context context) {
        if (operations == null) {
            operations = new Operations(context);
        }
        return operations;
    }

    private Operations(Context context) {
        this.context = context;
        this.database = new StonesDbHelper(context).getWritableDatabase();
    }



    private OperationsWrapper queryStones(String tableName) {

      Cursor cursor = database.query(tableName, null, null, null, null, null, null, null);

        return new OperationsWrapper(cursor);
    }

    public void updateStones(String name){

        OperationsWrapper cursorWrapper = queryStones(IF_TABLE);

        try{
            cursorWrapper.moveToFirst();
            while (!cursorWrapper.isAfterLast()){
                if(cursorWrapper.getInfinityStoneName().equals(name)) {
                    updateDbVisibility(name);
                    break;
                }

                cursorWrapper.moveToNext();
            }
        }finally {
            cursorWrapper.close();
        }

        cursorWrapper = queryStones(DETAIL_TABLE);

        try{
            cursorWrapper.moveToFirst();
            while (!cursorWrapper.isAfterLast()){
                if(cursorWrapper.getDetailStoneName().equals(name)) {
                    deleteDbStone(name);
                    break;
                }

                cursorWrapper.moveToNext();
            }
        }finally {
            cursorWrapper.close();
        }

    }


    public List<InfinityStone> getInfinityStoneList() {
        List<InfinityStone> list =  new ArrayList<>();

        OperationsWrapper cursorWrapper = queryStones(IF_TABLE);


        try{
            cursorWrapper.moveToFirst();
            while (!cursorWrapper.isAfterLast()){
                list.add(cursorWrapper.getInfinityStoneList());
                cursorWrapper.moveToNext();
            }
        }finally {
            cursorWrapper.close();
        }
        return list;

    }


    public List<SingleStone> getDetailStoneList() {
        List<SingleStone> list =  new ArrayList<>();

        OperationsWrapper cursorWrapper = queryStones(DETAIL_TABLE);


//        TODO all getters for each element to list
        try{
            cursorWrapper.moveToFirst();
            while (!cursorWrapper.isAfterLast()){
                list.add(cursorWrapper.getDetailStoneList());
                cursorWrapper.moveToNext();
            }
        }finally {
            cursorWrapper.close();
        }
        return list;

    }

    private void updateDbVisibility(String name){
        database.execSQL("UPDATE " + IF_TABLE + " SET " +
                COLUMN_STONE_VISIBILITY + " = 0 WHERE "
                + COLUMN_STONE_NAME + " = \"" + name + "\"");
    }

    private void deleteDbStone(String name){
        database.execSQL("DELETE FROM " + DETAIL_TABLE + " WHERE "
                + COLUMN_DETAIL_NAME + " = \"" + name + "\"");
    }


}

File: app/src/main/java/com/tkdev/ironmanstash/_LEGACY_infinity_stones/fragments/Operations.kt
package com.tkdev.ironmanstash._LEGACY_infinity_stones.fragments

import com.tkdev.ironmanstash._LEGACY_infinity_stones.fragments.allstones.InfinityStone
import com.tkdev.ironmanstash._LEGACY_infinity_stones.fragments.details.SingleStone
import android.content.Context
import android.database.sqlite.SQLiteDatabase
import com.tkdev.ironmanstash._LEGACY_infinity_stones.database.StonesContract
import com.tkdev.ironmanstash._LEGACY_infinity_stones.database.StonesContract.DetailStoneEntry
import com.tkdev.ironmanstash._LEGACY_infinity_stones.database.StonesContract.InfinityStonesEntry
import com.tkdev.ironmanstash._LEGACY_infinity_stones.database.StonesDbHelper
import java.util.*

class Operations internal constructor(private val context: Context) {
    private lateinit var operations: Operations

    private var database: SQLiteDatabase = StonesDbHelper(context).writableDatabase

    private fun queryStones(tableName: String): OperationsWrapper {
        val cursor = database.query(tableName, null, null, null, null, null, null, null)
        return OperationsWrapper(cursor)
    }

    fun updateStones(name: String) {
        var cursorWrapper = queryStones(StonesContract.IF_TABLE)
        try {
            cursorWrapper.moveToFirst()
            while (!cursorWrapper.isAfterLast) {
                if (cursorWrapper.infinityStoneName == name) {
                    updateDbVisibility(name)
                    break
                }
                cursorWrapper.moveToNext()
            }
        } finally {
            cursorWrapper.close()
        }
        cursorWrapper = queryStones(StonesContract.DETAIL_TABLE)
        try {
            cursorWrapper.moveToFirst()
            while (!cursorWrapper.isAfterLast) {
                if (cursorWrapper.detailStoneName == name) {
                    deleteDbStone(name)
                    break
                }
                cursorWrapper.moveToNext()
            }
        } finally {
            cursorWrapper.close()
        }
    }

     val infinityStoneList: List<Any>
        get() {
            val list: MutableList<InfinityStone> = ArrayList<InfinityStone>()
            val cursorWrapper = queryStones(StonesContract.IF_TABLE)
            cursorWrapper.use { cursorWrapper ->
                cursorWrapper.moveToFirst()
                while (!cursorWrapper.isAfterLast) {
                    list.add(cursorWrapper.infinityStoneList)
                    cursorWrapper.moveToNext()
                }
            }
            return list
        }

     val detailStoneList: List<Any>
        get() {
            val list: MutableList<SingleStone> = ArrayList<SingleStone>()
            val cursorWrapper = queryStones(StonesContract.DETAIL_TABLE)
            cursorWrapper.use { cursorWrapper ->
                cursorWrapper.moveToFirst()
                while (!cursorWrapper.isAfterLast) {
                    list.add(cursorWrapper.detailStoneList)
                    cursorWrapper.moveToNext()
                }
            }
            return list
        }

    private fun updateDbVisibility(name: String) {
        database.execSQL("UPDATE " + StonesContract.IF_TABLE + " SET " +
                InfinityStonesEntry.COLUMN_STONE_VISIBILITY + " = 0 WHERE "
                + InfinityStonesEntry.COLUMN_STONE_NAME + " = \"" + name + "\"")
    }

    private fun deleteDbStone(name: String) {
        database.execSQL("DELETE FROM " + StonesContract.DETAIL_TABLE + " WHERE "
                + DetailStoneEntry.COLUMN_DETAIL_NAME + " = \"" + name + "\"")
    }

    fun gauntletVisibility(): Boolean {
        return detailStoneList.isEmpty()
    }

}